### PR TITLE
[DO NOT MERGE] Bully Celt on his bad code

### PR DESCRIPTION
**Pull request by ghost**
This is solely for you to bully my bad code and insult my cockups as I work on the vending machine overhaul. This is not to be merged because it's nowhere near complete.

DONE:
- Made coinboxes undestroyable and unbreakable because I'm going to make items be stored inside this and it would be a dick move (and something people'd do) if people came around and stole the entire contents of your machine
- ATMs now have an option to print a coinbox & vending machine circuitboard in a neat box with a little cute receipt (http://i.imgur.com/7hREPJG.png)
- because the fucker who coded those cute cargo boxes didn't make it a storage item, made it able to store/empty more than one item in it.
- Added Product Boxes
  - now these things contain all of the stuff that is in the vending machine
  - they can be DNA locked
- fixed a few problems

TODO:
a todo list
and err stocking stuff. I'll probably steal all of the smartfridge things.

### CODE REVIEW

**Comment by d3athrow** [comment]
You can avoid the merge commits by using feature branches and rebasing. http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by DrCelt** [comment]
@d3athrow 
thanks pomfykins I'll look into that then!

File: code/WorkInProgress/Mini/ATM.dm [hunk 1]
```diff
@@ -340,7 +341,7 @@ log transactions
 						usr << "<span class='notice'>The [src.name] flashes an error on its display.</span>"
 						return
 					lastprint = world.timeofday
-					var/obj/item/weapon/paper/R = new(src.loc)
+					var/obj/item/weapon/paper/R = new /obj/item/weapon/paper(src.loc)
```
Comment (ComicIronic) [inline code/WorkInProgress/Mini/ATM.dm]: This isn't required, which is why it wasn't in the original code.

File: code/WorkInProgress/Mini/ATM.dm [hunk 2]
```diff
@@ -362,6 +363,52 @@ log transactions
 					playsound(loc, 'sound/items/polaroid1.ogg', 50, 1)
 				else
 					playsound(loc, 'sound/items/polaroid2.ogg', 50, 1)
+			if("vending_machine")
+				if(authenticated_account)
+					if(world.timeofday < lastprint + PRINT_DELAY)
+						usr << "<span class='notice'>The [src.name] flashes an error on its display.</span>"
```
Comment (ComicIronic) [inline code/WorkInProgress/Mini/ATM.dm]: Make this more clear: it should mention the print delay.

File: code/game/objects/items/weapons/storage/lockbox.dm [hunk 3]
```diff
@@ -74,27 +75,28 @@
...
-	var/newsev = max(3,severity+1)
-	for(var/atom/movable/A as mob|obj in src)//pulls everything out of the locker and hits it with an explosion
-		A.loc = src.loc
-		A.ex_act(newsev)
-	newsev=4-severity
-	if(prob(newsev*25)+25) // 1=100, 2=75, 3=50
-		qdel(src)
+	if (!src.indestructible)
+		var/newsev = max(3,severity+1)
+		for(var/atom/movable/A as mob|obj in src)//pulls everything out of the locker and hits it with an explosion
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/storage/lockbox.dm]: You don't need (as mob|obj)
That's a term only ever used for verbs and user inputs.

File: code/modules/recycling/sortingmachinery.dm [hunk 4]
```diff
@@ -57,14 +57,22 @@
...
+	var/turf/T = get_turf(src)
+	for(var/obj/item/I in contents)
+		if(T)
+			var/mob/M
+			if(ismob(loc))
+				M = loc
+				I.dropped(M)
+			if(ismob(T))
+				M = T
+				I.pickup(M)
```
Comment (ComicIronic) [inline code/modules/recycling/sortingmachinery.dm]: You shouldn't need this, because that's what put_in_active_hand is for.

**Comment by ComicIronic** [comment]
>make items stored inside coinboxes

Not gonna lie, I think that's going to run you into some issues.
Coinboxes are used for storing coins and vouchers. If you have items in the coinbox able to be sold, you could just buy back a coin.

File: code/WorkInProgress/Mini/ATM.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/WorkInProgress/Mini/ATM.dm]: Also
>usr
Not safe.

File: code/modules/recycling/sortingmachinery.dm (hunk 4 above)
Comment (DrCelt) [inline code/modules/recycling/sortingmachinery.dm]: @comicironic
This is to ensure that it actually spills out its contents, previously it just spit one thing in the variable "Wrapped".

File: code/game/objects/items/weapons/storage/lockbox.dm (hunk 3 above)
Comment (DrCelt) [inline code/game/objects/items/weapons/storage/lockbox.dm]: @ComicIronic
I just added an if statement, that was originally in the code. Should I remove it in the next revision?

File: code/WorkInProgress/Mini/ATM.dm (hunk 2 above)
Comment (DrCelt) [inline code/WorkInProgress/Mini/ATM.dm]: @ComicIronic
Yeah, that was a problem I'd thought about too. It was in the original ATM code. I'll make it more clear.
@Iamgoofball
I'm fairly new to byond code; why is it unsafe and how can I make it safe?

File: code/WorkInProgress/Mini/ATM.dm (hunk 1 above)
Comment (DrCelt) [inline code/WorkInProgress/Mini/ATM.dm]: @ComicIronic
Thanks; I'll change that back.

**Comment by DrCelt** [comment]
@ComicIronic
The way I've worked it out in my head should result in no issues, but don't hold me to it. I'll come to that issue when it comes.

You don't mind me bastardising its use though?

**Comment by Midaychi** [comment]
@ComicIronic Would be neat to use that for bartering-based vending machines. Problem would be either coming up with an in-depth interface for defining trade value or having an automated system that can somehow tell that a goat in the coinbox is worth a hundred plasmacoins or something.

File: code/WorkInProgress/Mini/ATM.dm (hunk 1 above)
Comment (clusterfack) [inline code/WorkInProgress/Mini/ATM.dm]: Its a protip, if the type is defined in the variable and you just call new(location) like that, it takes the type of the variable!

File: code/modules/recycling/sortingmachinery.dm (hunk 4 above)
Comment (DrCelt) [inline code/modules/recycling/sortingmachinery.dm]: Also there are three items in the package so you wouldn't have enough hands for it all.

File: code/WorkInProgress/Mini/ATM.dm (hunk 1 above)
Comment (DrCelt) [inline code/WorkInProgress/Mini/ATM.dm]: Right, thanks! I'll keep that in mind for future

File: code/game/objects/items/weapons/storage/lockbox.dm (hunk 3 above)
Comment (ComicIronic) [inline code/game/objects/items/weapons/storage/lockbox.dm]: Yes, there's no reason for it to be there.

File: code/WorkInProgress/Mini/ATM.dm (hunk 1 above)
Comment (ComicIronic) [inline code/WorkInProgress/Mini/ATM.dm]: You don't need to change it.
All the code in this section uses usr, and it's a Topic, so it's pretty much standard.

File: code/WorkInProgress/Mini/ATM.dm (hunk 2 above)
Comment (ComicIronic) [inline code/WorkInProgress/Mini/ATM.dm]: You don't need to change usr.
All the code in this section uses usr, and it's a Topic, so it's pretty much standard.

File: code/game/machinery/constructable_frame.dm [hunk 5]
```diff
@@ -858,12 +858,10 @@ obj/item/weapon/circuitboard/rdserver
 	name = "Circuit Board (Vending Machine)"
 	build_path = "/obj/machinery/vending"
 	board_type = "machine"
-	origin_tech = "materials=1;engineering=1;powerstorage=1"
-	frame_desc = "Requires 1 Matter Bins, 1 Scanning Module, and 1 Manipulator."
+	origin_tech = "	materials=1;engineering=1;powerstorage=1"
```
Comment (ComicIronic) [inline code/game/machinery/constructable_frame.dm]: Randomspacewhy

File: code/game/machinery/vending.dm [hunk 6]
```diff
@@ -85,10 +85,7 @@
 	overlays_vending[1] = "[icon_state]-panel"
 
 	component_parts = newlist(\
-		/obj/item/weapon/circuitboard/vendomat,\
-		/obj/item/weapon/stock_parts/matter_bin,\
-		/obj/item/weapon/stock_parts/manipulator,\
-		/obj/item/weapon/stock_parts/scanning_module\
+		/obj/item/weapon/storage/lockbox/coinbox
```
Comment (ComicIronic) [inline code/game/machinery/vending.dm]: You have to set the coinbox var to this, don't forget.

File: code/game/machinery/vending.dm [hunk 7]
```diff
@@ -85,10 +85,7 @@
 	overlays_vending[1] = "[icon_state]-panel"
 
 	component_parts = newlist(\
-		/obj/item/weapon/circuitboard/vendomat,\
```
Comment (ComicIronic) [inline code/game/machinery/vending.dm]: Don't remove this.

File: code/game/machinery/vending.dm [hunk 8]
```diff
@@ -805,11 +797,7 @@
 	products = list(/obj/item/weapon/reagent_containers/food/drinks/coffee = 25,/obj/item/weapon/reagent_containers/food/drinks/tea = 25,/obj/item/weapon/reagent_containers/food/drinks/h_chocolate = 25)
 	contraband = list(/obj/item/weapon/reagent_containers/food/drinks/ice = 10)
 	prices = list(/obj/item/weapon/reagent_containers/food/drinks/coffee = 25, /obj/item/weapon/reagent_containers/food/drinks/tea = 25, /obj/item/weapon/reagent_containers/food/drinks/h_chocolate = 25)
-
```
Comment (ComicIronic) [inline code/game/machinery/vending.dm]: Why would you remove all these lines they were there for readability
This file is hard enough to read without everything being in one big block

File: code/modules/recycling/sortingmachinery.dm (hunk 4 above)
Comment (ComicIronic) [inline code/modules/recycling/sortingmachinery.dm]: >This is to ensure that it actually spills out its contents

I can't see why that would be the case. Not only does it call pickup in a case where the mob might not pick up the item, the default pickup behaviour does nothing. No reason to have it.

File: code/game/machinery/vending.dm (hunk 8 above)
Comment (DrCelt) [inline code/game/machinery/vending.dm]: They removed themselves and I'm not sure why.

File: code/game/machinery/vending.dm (hunk 7 above)
Comment (DrCelt) [inline code/game/machinery/vending.dm]: Alright

File: code/game/machinery/vending.dm (hunk 6 above)
Comment (DrCelt) [inline code/game/machinery/vending.dm]: Thanks, will do

File: code/game/machinery/constructable_frame.dm (hunk 5 above)
Comment (DrCelt) [inline code/game/machinery/constructable_frame.dm]: oh fuck, yeah. thanks.

File: code/game/machinery/vending.dm (hunk 8 above)
Comment (ComicIronic) [inline code/game/machinery/vending.dm]: What are you using to edit with?

Comment (DrCelt) [inline code/game/machinery/vending.dm]: byond

File: code/game/machinery/vending.dm (hunk 6 above)
Comment (DrCelt) [inline code/game/machinery/vending.dm]: Right, how exactly do I do that..?

File: code/WorkInProgress/Mini/ATM.dm (hunk 1 above)
Comment (DrCelt) [inline code/WorkInProgress/Mini/ATM.dm]: This has been fixed for the next revision.

File: code/WorkInProgress/Mini/ATM.dm (hunk 2 above)
Comment (DrCelt) [inline code/WorkInProgress/Mini/ATM.dm]: This has been fixed for the next revision.

File: code/game/machinery/vending.dm (hunk 7 above)
Comment (DrCelt) [inline code/game/machinery/vending.dm]: Fixed for the next revision

File: code/game/machinery/vending.dm (hunk 6 above)
Comment (DrCelt) [inline code/game/machinery/vending.dm]: Fixed for the next revision.

File: code/game/objects/items/weapons/storage/lockbox.dm (hunk 3 above)
Comment (DrCelt) [inline code/game/objects/items/weapons/storage/lockbox.dm]: fixed next revision

File: code/modules/recycling/sortingmachinery.dm (hunk 4 above)
Comment (DrCelt) [inline code/modules/recycling/sortingmachinery.dm]: I'm not sure how you'd like me to fix this, sorry. Mind elaborating?

File: code/game/machinery/constructable_frame.dm (hunk 5 above)
Comment (DrCelt) [inline code/game/machinery/constructable_frame.dm]: Not fixed in the latest revision. I'll do that now.

**Comment by DrCelt** [comment]
I used the word "instead". It should've been "alongside."
sorry.

File: code/game/objects/items/weapons/storage/lockbox.dm (hunk 3 above)
Comment (DrCelt) [inline code/game/objects/items/weapons/storage/lockbox.dm]: Or maybe not. I did fix the one above this one though, I guess that's where the confusion came from.

File: code/modules/recycling/sortingmachinery.dm (hunk 4 above)
Comment (ComicIronic) [inline code/modules/recycling/sortingmachinery.dm]: Just remove I.pickup(M)

**Comment by DrCelt** [comment]
what does a branch failed check mean
should I panic

**Comment by ComicIronic** [comment]
You managed do make something that didn't compile.

**Comment by DrCelt** [comment]
Great.
By compile it means byond compile?

File: code/game/objects/items/weapons/storage/lockbox.dm [hunk 9]
```diff
@@ -56,15 +59,34 @@
...
 				O.show_message(text("<span class='notice'>The locker has been broken by [] with an electromagnetic card!</span>", user), 1, text("You hear a faint electrical spark."), 2)
 
+	else if (canbednalocked && istype(W, /obj/item/device/multitool))
+		if ((!dnalocked)|| (src.dnastring == user.dna.unique_enzymes))
+			if (src.dnalocked == 0)
+				user << "<span class='rose'>The lockbox scans your fingerprint and saves it for later.</span>"
+				src.dnastring = user.dna.unique_enzymes
+				src.dnalocked = 1
+				tracked_access = "This Lockbox is registered to: [dnastring]"
+			locked = !locked
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/storage/lockbox.dm]: Can't do this and then have the else if

**Comment by ComicIronic** [comment]
Yes.

**Comment by DrCelt** [comment]
@ComicIronic 
Odd. Let me try that again, then.

File: code/game/objects/items/weapons/storage/lockbox.dm (hunk 9 above)
Comment (DrCelt) [inline code/game/objects/items/weapons/storage/lockbox.dm]: Why not? I'm not sure I understand what you mean

Comment (DrCelt) [inline code/game/objects/items/weapons/storage/lockbox.dm]: Oh, I see the issue.

**Comment by DrCelt** [comment]
Oh, I see the issue.

**Comment by DrCelt** [comment]
t-thanks travis

File: code/modules/recycling/sortingmachinery.dm (hunk 4 above)
Comment (DrCelt) [inline code/modules/recycling/sortingmachinery.dm]: Fixed next revision

File: code/game/machinery/constructable_frame.dm (hunk 5 above)
Comment (DrCelt) [inline code/game/machinery/constructable_frame.dm]: Fixed latest revision

File: code/game/objects/items/weapons/storage/lockbox.dm (hunk 3 above)
Comment (DrCelt) [inline code/game/objects/items/weapons/storage/lockbox.dm]: Fixed in next revision.

**Comment by SkowronX** [comment]
Any breakthroughs?

**Comment by DrCelt** [comment]
@SkowronX
Still working on integrating smart fridge stuff which will be a big commit
but slow and steady wins the race, eh?

**Comment by SkowronX** [comment]
To a point, until you have to deal with that lovely grey icon down there that says "merge conflicts".

**Comment by DrCelt** [comment]
C'est la vie

**Comment by DrCelt** [comment]
Ah bollocks I did the thing with the map you're not meant to do

**Comment by DrCelt** [comment]
one moment

**Comment by d3athrow** [comment]
Reopen when you get this figured out.

**Pull request closed without merging**